### PR TITLE
test(node-bindings): add argument count unit tests

### PR DESCRIPTION
**Pull request by matthewkeil**
Relates to audit comments.

There was an audit concern about checking argument counts passed to functions.  Node.js helps to ensure argument correctness.  Added unit tests to demonstrate error handling.

### Reviewer Notes
- Followed spec test format for "valid" case (see line [L#118-L#163](https://github.com/matthewkeil/c-kzg-4844/blob/2fb362c298afe7c4088973289d509566fa39cd4a/bindings/node.js/test/kzg.test.ts#L118) for an example)
- Used random valid test from main `tests` folder for each case to ensure base case correctness

### CODE REVIEW

File: bindings/node.js/test/kzg.test.ts [hunk 1]
```diff
@@ -81,6 +81,35 @@ function bytesEqual(a: Uint8Array | Buffer, b: Uint8Array | Buffer): boolean {
...
+  const validTest = tests.find(
+    (testFile: string) =>
+      !testFile.includes("invalid") && !testFile.includes("incorrect") && !testFile.includes("different")
+  );
+  if (!validTest) throw new Error("Could not find valid test");
+  return yaml.load(readFileSync(validTest, "ascii"));
+}
+
+function testArgCount(fn: (...args: any[]) => any, validArgs: any[]): void {
+  const lessArgs = validArgs.length === 1 ? [] : validArgs.slice(0, -1);
```
Comment (g11tech) [inline bindings/node.js/test/kzg.test.ts:95]:
```suggestion
  const lessArgs = validArgs.slice(0, -1);
```
e.g. `[1].slice(0,-1)` is already `[]`

File: bindings/node.js/test/kzg.test.ts [hunk 2]
```diff
@@ -81,6 +81,35 @@ function bytesEqual(a: Uint8Array | Buffer, b: Uint8Array | Buffer): boolean {
...
+  expect(lessArgs.length).toBeLessThan(validArgs.length);
+  const moreArgs = validArgs.concat("UNKNOWN_ARGUMENT");
+  expect(moreArgs.length).toBeGreaterThan(validArgs.length);
+  it("should run for expected argument count", () => {
```
Comment (g11tech) [inline bindings/node.js/test/kzg.test.ts:99]: space between each `it`
```suggestion

  it("should run for expected argument count", () => {
```

File: bindings/node.js/test/kzg.test.ts [hunk 3]
```diff
@@ -81,6 +81,35 @@ function bytesEqual(a: Uint8Array | Buffer, b: Uint8Array | Buffer): boolean {
   return true;
 }
 
+function getValidTest(testDir: string): any {
```
Comment (g11tech) [inline bindings/node.js/test/kzg.test.ts:84]: jsdoc: `Helper function to return first valid test in testDir`

File: bindings/node.js/test/kzg.test.ts [hunk 4]
```diff
@@ -234,19 +263,29 @@ describe("C-KZG", () => {
   });
 
   describe("edge cases for blobToKzgCommitment", () => {
+    describe("check argument count", () => {
```
Comment (g11tech) [inline bindings/node.js/test/kzg.test.ts:266]: space out
```suggestion

    describe("check argument count", () => {
```

File: bindings/node.js/test/kzg.test.ts (hunk 1 above)
Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts:95]: Good catch. Updated.

File: bindings/node.js/test/kzg.test.ts (hunk 2 above)
Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts:99]: Updated.

File: bindings/node.js/test/kzg.test.ts (hunk 4 above)
Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts:273]: The lint rules prevent this change.  I added spaces between `describe` and `it` blocks where possible and ran lint.  Please verify things look acceptable.

File: bindings/node.js/test/kzg.test.ts [hunk 5]
```diff
@@ -81,6 +81,35 @@ function bytesEqual(a: Uint8Array | Buffer, b: Uint8Array | Buffer): boolean {
...
+  const tests = globSync(testDir);
+  const validTest = tests.find(
+    (testFile: string) =>
+      !testFile.includes("invalid") && !testFile.includes("incorrect") && !testFile.includes("different")
+  );
+  if (!validTest) throw new Error("Could not find valid test");
+  return yaml.load(readFileSync(validTest, "ascii"));
+}
+
+function testArgCount(fn: (...args: any[]) => any, validArgs: any[]): void {
```
[the remark this answers is not in the document]
Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts:120]: Updated.

File: bindings/node.js/test/kzg.test.ts (hunk 3 above)
Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts:95]: Updated.

File: bindings/node.js/test/kzg.test.ts [hunk 6]
```diff
@@ -53,32 +67,101 @@ const generateRandomBlob = (): Uint8Array => {
...
+ * Verifies that two Uint8Arrays are bitwise equivalent
+ *
+ * @param {Uint8Array} a
+ * @param {Uint8Array} b
+ *
+ * @return {void}
+ *
+ * @throws {Error} If arrays are not equal length or byte values are unequal
+ */
+function bytesEqual(a: Uint8Array | Buffer, b: Uint8Array | Buffer): void {
```
Comment (g11tech) [inline bindings/node.js/test/kzg.test.ts:96]:
```suggestion
function assertEqual(a: Uint8Array | Buffer, b: Uint8Array | Buffer): void {
```
if we throw then its better to rename it as `assert` as its semantics are more clear that way

Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts:96]: Updated.

Comment (matthewkeil) [reply bindings/node.js/test/kzg.test.ts]: Updated.

**Review by g11tech — APPROVED** [review]
lgtm

**Comment by matthewkeil** [comment]
@asn-d6, I am curious if there is anything stopping this from approval?  I will be happy to update if there are any concerns remaining.

**Review by asn-d6 — APPROVED** [review]
LGTM!

I'm merging this even though the Rust tests are failing, since there is clearly no relation to this PR and the rust bindings.

**Comment by jimmygchen** [comment]
> LGTM!
> 
> I'm merging this even though the Rust tests are failing, since there is clearly no relation to this PR and the rust bindings.

Looks like this is related to the new Rust stable version v1.70 released on 1st June.

Last passing build using Rust had v1.69
https://github.com/ethereum/c-kzg-4844/actions/runs/5147246203/jobs/9267396040

All newer builds are on Rust v1.70 and failed.

**Comment by asn-d6** [comment]
Thanks all! Merging!
